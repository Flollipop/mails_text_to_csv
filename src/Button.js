import * as PropTypes from 'prop-types';
import React from 'react';

const buttonStyle = {
  display: 'flex',
  maxHeight: '40px',
  backgroundColor: '#01A7CA',
  color: 'white',
  border: 0,
};

export function Button(props) {
  return (
    <button style={buttonStyle} onClick={props.onClick}>
      convert
    </button>
  );
}

Button.propTypes = { onClick: PropTypes.func };
