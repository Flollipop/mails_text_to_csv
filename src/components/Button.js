import * as PropTypes from 'prop-types';
import React from 'react';

export const buttonTestId = 'button';

export function Button({ onClick, label, disabled }) {
  return (
    <button data-testid={buttonTestId} onClick={onClick} disabled={disabled}>
      {label}
    </button>
  );
}

Button.defaultProps = {
  disabled: false,
};

Button.propTypes = { onClick: PropTypes.func, label: PropTypes.string, disabled: PropTypes.bool };
