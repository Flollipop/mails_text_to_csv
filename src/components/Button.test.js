import React from 'react';
import ReactDOM from 'react-dom';
import { Button, buttonTestId } from './Button';
import { render, screen, fireEvent, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

afterEach(cleanup);

test('render button - smoke test', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Button />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('renders button test label', () => {
  const labelToTest = 'button label';
  const { getByTestId } = render(<Button label={labelToTest} />);
  expect(getByTestId(buttonTestId)).toHaveTextContent(labelToTest);
});

test('renders button test onClick', () => {
  const onClick = jest.fn();
  const labelToTest = 'button label';
  const { getByTestId } = render(<Button label={labelToTest} onClick={onClick} />);

  expect(getByTestId(buttonTestId)).toBeTruthy();
  fireEvent.click(screen.getByTestId(buttonTestId));
  expect(onClick).toHaveBeenCalledTimes(1);
});

test('renders disabled button test no onClick call', () => {
  const onClick = jest.fn();
  const labelToTest = 'button label';
  const { getByTestId } = render(<Button label={labelToTest} onClick={onClick} disabled={true} />);

  expect(getByTestId(buttonTestId)).toBeTruthy();
  fireEvent.click(screen.getByTestId(buttonTestId));
  expect(onClick).toHaveBeenCalledTimes(0);
});

test('button snapshot', () => {
  const tree = renderer.create(<Button label='button' />).toJSON();
  expect(tree).toMatchSnapshot();
});
