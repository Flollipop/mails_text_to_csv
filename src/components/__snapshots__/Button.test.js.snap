// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button snapshot 1`] = `
<button
  data-testid="button"
  disabled={false}
>
  button
</button>
`;
